### PR TITLE
Add missing test policy waits

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
/kind test

**What this PR does / why we need it**:
There were some missing policy condition wait this create
some race conditions at handler e2e test, this also add
two functions with the `AndWait` suffix to do the update + wait
at one line.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
/retest

prow hiccup
```
/tmp/knmstate/kubernetes-nmstate/build/_output/bin//go//bin//operator-sdk build quay.io/nmstate/kubernetes-nmstate-handler --image-builder docker
time="2020-05-06T08:10:12Z" level=info msg="Building OCI image quay.io/nmstate/kubernetes-nmstate-handler"
time="2020-05-06T08:10:26Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: connection refused"
context canceled
Error: failed to output build image quay.io/nmstate/kubernetes-nmstate-handler: failed to exec []string{"docker", "build", "-f", "build/Dockerfile", "-t", "quay.io/nmstate/kubernetes-nmstate-handler", "."}: exit status 1

```

File: test/e2e/handler/nnce_desiredstate_test.go [hunk 1]
```diff
@@ -6,27 +6,18 @@ import (
...
 			By("Create a policy")
-			updateDesiredState(linuxBrUp(bridge1))
-			policyConditionsStatusEventually().Should(ContainElement(
-				nmstatev1alpha1.Condition{
-					Type:   nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
-					Status: corev1.ConditionTrue,
-				},
-			))
+			updateDesiredStateAndWait(linuxBrUp(bridge1))
 		})
```
Comment (yboaron) [inline test/e2e/handler/nnce_desiredstate_test.go:17]: Is this code checks exactly what waitForAvailableTestPolicy() does ?

File: test/e2e/handler/nncp_cleanup_test.go [hunk 2]
```diff
@@ -15,24 +15,24 @@ import (
...
+			By("Wait for policy to be ready")
+			waitForAvailablePolicy(bridge1)
 
-	AfterEach(func() {
-		updateDesiredState(linuxBrAbsent(bridge1))
-		waitForAvailableTestPolicy()
-		resetDesiredStateForNodes()
-	})
+			By("Delete the policy")
+			deletePolicy(bridge1)
```
Comment (yboaron) [inline test/e2e/handler/nncp_cleanup_test.go:27]: Why do you need to call deletePolicy in both BeforeEach and AfterEach?

Comment (qinqon) [reply test/e2e/handler/nncp_cleanup_test.go:27]: In case something fails AfterEach will remove it (or try to)

Comment (qinqon) [reply test/e2e/handler/nncp_cleanup_test.go:27]: In case something goes sideways at BeforeEach and deletePolicy is not reached AfterEach will clean it.

File: test/e2e/handler/utils.go [hunk 3]
```diff
@@ -165,6 +165,9 @@ func deletePolicy(name string) {
 	policy := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
 	policy.Name = name
 	err := framework.Global.Client.Delete(context.TODO(), policy)
+	if apierrors.IsNotFound(err) {
```
Comment (yboaron) [inline test/e2e/handler/utils.go:168]: I know it's a corner case, but don't we want to be on the safe side and verify also that NodeNetworkConfigurationEnactment was deleted although policy not found?

File: test/e2e/handler/webhook_test.go [hunk 4]
```diff
@@ -18,12 +18,10 @@ var _ = Describe("Mutating Admission Webhook", func() {
 			// Make sure test policy is not there so
 			// we exercise CREATE event
 			resetDesiredStateForNodes()
-			updateDesiredState(linuxBrUp(bridge1))
+			updateDesiredStateAndWait(linuxBrUp(bridge1))
 		})
 		AfterEach(func() {
-			waitForAvailableTestPolicy()
```
Comment (yboaron) [inline test/e2e/handler/webhook_test.go:24]: Don't we need the first waitForAvailableTestPolicy() ?

**Comment by yboaron** [comment]
looks fine, just a few Qs inside

File: test/e2e/handler/utils.go (hunk 3 above)
Comment (qinqon) [reply test/e2e/handler/utils.go:168]: That's super good question, we know we are missing adding the block owner deletetion flag [1] at [2] but we will do that at different PR, I think

[1] ownerReference.blockOwnerDeletion=true-> https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion

[2] https://github.com/nmstate/kubernetes-nmstate/blob/fad730940ba634da99785b3dc5d99caf696c70af/pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationenactment_types.go#L76

File: test/e2e/handler/webhook_test.go (hunk 4 above)
Comment (qinqon) [reply test/e2e/handler/webhook_test.go:24]: We have move the wait to the proper places I think we don't need this anymore here right ? this test know just check that the label is there we don't care if it's after Status is good (for webhook label is set after we finish applying the manifests but wait is good to be sure we are at finished status).

File: test/e2e/handler/utils.go (hunk 3 above)
Comment (qinqon) [reply test/e2e/handler/utils.go:168]: @yboaron I have create a PR for that https://github.com/nmstate/kubernetes-nmstate/pull/526

**Comment by yboaron** [comment]
Tests are green, I can LGTM
